crates/compiler/builtins/roc/Str.roc

```diff
@@ -1052,3 +1055,37 @@ contains = \haystack, needle ->
     when firstMatch haystack needle is
         Some _index -> Bool.true
         None -> Bool.false
+
+## Drops the given prefix [Str] from the start of a [Str]
+## If the prefix is not found, returns the original string.
+##
+## ```roc
+## expect Str.dropPrefix "bar" "foo" == "bar"
+## expect Str.dropPrefix "foobar" "foo" == "bar"
+## ```
+dropPrefix : Str, Str -> Str
+dropPrefix = \haystack, prefix ->
+    if Str.startsWith haystack prefix then
+        start = Str.countUtf8Bytes prefix
+        len = Num.subSaturated (Str.countUtf8Bytes haystack) start
```

Comment:
Isn't this subtraction guaranteed to never exercise the saturating behavior since `start` = `Str.countUtf8Bytes prefix` <= `Str.countUtf8Bytes haystack` since `Str.startsWith haystack prefix`? It would probably be more performant to use wrapping subtraction.